sessionctx/variable/varsutil.go

```diff
@@ -737,3 +738,31 @@ func setAnalyzeTime(s *SessionVars, val string) (string, error) {
 	}
 	return t.Format(AnalyzeFullTimeFormat), nil
 }
+
+// serverGlobalVariable is used to handle variables that acts in server and global scope.
+type serverGlobalVariable struct {
+	sync.Mutex
+	serverVal string
+	globalVal string
+}
+
+// Set sets the value according to variable scope.
+func (v *serverGlobalVariable) Set(val string, isServer bool) {
+	v.Lock()
+	if isServer {
+		v.serverVal = val
+	} else {
+		v.globalVal = val
+	}
+	v.Unlock()
+}
+
+// GetVal gets the value.
+func (v *serverGlobalVariable) GetVal() string {
+	v.Lock()
```

Comment:
Can we just do read lock here?

Reply:
I think it is not necessary because we do not expect the server global variables be set or read very frequent, a `mutex` is enough now.